app/src/lib/git/spawn.ts

```diff
@@ -132,6 +132,7 @@ export async function spawnAndComplete(
         })
         return
       } else {
+        log.error(stderr.toString())
```

Comment:
I'm a bit hesitant about this, mainly because there are no other logs in this file, and this log doesn't give any context about when or how was that `stderr` obtained. Maybe it's better to add it to the description of the `Error` in the `reject`?

Reply:
@niik you probably know better the whole chain of error handlers. What do you think about this ☝️ ?

Another idea is adding the `stderr` as the `cause` of the error:
`throw new Error('blah blah blah', { cause: stderr })`

Reply:
FWIW, I pulled this and tested (just by modifying the if to always fall to the reject, so not a real error scenario) to what it would look like for this to happen. I checked that we use `log.error` in other `lib/git` files.  We have a couple other places where we just `log.error(error)` in conjunction with throwing a more human readable error. My thought is that the human readable error would lead to checking the logs and you would now see the stderr output. 

That said, adding a preface to the error would make it even better and we do that way more than not in the code base. I think just something like `Error in spawnAndComplete, stderr: ${stdErr}` seems good here. 

I believe the code is working on the assumption that the any expected error would have been handled by the caller and no more context is needed (as the error given states :D ) and probably why we didn't think to log out the error before hand, but the error this user received here was due to a faulty environment  - not something we would usually expect to handle in a way other than to just log errors so that can be determined.
 
<img width="1471" alt="image" src="https://user-images.githubusercontent.com/75402236/216321424-c3bda70d-15ef-4b9a-bd2d-b8c0c6181b82.png">

Reply:
TLDR: I think either log like pr suggests (adding prefacing as Sergio suggested) or go ahead and append to the dialog error since this scenario is very unexpected and you would want to go look for this extra context anyways.

Reply:
> That said, adding a preface to the error would make it even better and we do that way more than not in the code base. I think just something like `Error in spawnAndComplete, stderr: ${stdErr}` seems good here.

I added that in the preface :

![image](https://user-images.githubusercontent.com/14168019/216348776-b1a00bd4-d1d6-4739-a3c6-80f8c11a2a24.png)

Reply:
> @niik you probably know better the whole chain of error handlers. What do you think about this ☝️ ?

Honestly I think we should get this in line with the errors produced from the core.ts/git function so that it raises GitErrors with the rawMessage set to true for cases where we can't map the error to something we know could be going wrong (too large output for example). I know I've looked at this before but I can't remember why I didn't just do it then, either I got pulled away to something else or there was some piece of this that lived in dugite that was hard to extract. But we shouldn't have different error paths depending on whether Git is invoked through core.ts or spawn.ts.

Reply:
I don't really know how I would implement it, I'm not very aware how on the whole program works